assessments/projects/rook/self-assessment.md

```diff
@@ -0,0 +1,374 @@
+validates that the request corresponds to a file the client is authorized to 
+access, and then returns a token to the client.  The client then transmits that 
+token to the file server, which, after confirming its validity, returns the file.
+
+Rook can be used to automatically configure the Ceph CSI drivers to mount the storage to an application's pods. See image at top of [Actors](#actors) for reference
+
+##### Actors
+
+* Rook Operator
+* Ceph CSI Drivers
+* Ceph Daemons
+
+#### Workflows
+
+##### Block Storage Example
+
+In the diagram [here](#actors), the flow to create an application with an RWO volume is:
+
+1. The (blue) app creates a PVC to request storage
+2. The PVC defines the Ceph RBD storage class (sc) for provisioning the storage
+3. K8s calls the Ceph-CSI RBD provisioner to create the Ceph RBD image.
+4. The kubelet calls the CSI RBD volume plugin to mount the volume in the app
+5. The volume is now available for reads and writes.
+
+A ReadWriteOnce volume can be mounted on one node at a time.
+
+##### Shared File System Example
+
+In the diagram [here](#actors), the flow to create a applications with a RWX volume is:
+
+1. The (purple) app creates a PVC to request storage
+2. The PVC defines the CephFS storage class (sc) for provisioning the storage
+3. K8s calls the Ceph-CSI CephFS provisioner to create the CephFS subvolume
+4. The kubelet calls the CSI CephFS volume plugin to mount the volume in the app
+5. The volume is now available for reads and writes.
+
+A ReadWriteMany volume can be mounted on multiple nodes for your application to use.
+
+##### Block Storage Example
+
+In the diagram [here](#actors), the flow to create an application with access to an S3 bucket is:
+
+- The (orange) app creates an ObjectBucketClaim (OBC) to request a bucket
+- The Rook operator creates a Ceph RGW bucket (via the lib-bucket-provisioner)
+- The Rook operator creates a secret with the credentials for accessing the bucket and a configmap with bucket information
+- The app retrieves the credentials from the secret
+- The app can now read and write to the bucket with an S3 client
+
+A S3 compatible client can use the S3 bucket right away using the credentials (`Secret`) and bucket info (`ConfigMap`).
+
+
+### Goals
+
+####
+Rook automates deployment and management of Ceph to provide self-managing, self-scaling, and self-healing storage services. The Rook operator does this by building on Kubernetes resources to deploy, configure, provision, scale, upgrade, and monitor Ceph.
+
+#### Security Goals
+
+* All access to rook operator should be authenticated and authorized
+* Secrets created by Rook operator should maintain confidentiality
+* Rook operator should obey the principle of least privilege (see [here](https://github.com/rook/rook/pull/1736) for improvements already made to operator permissions model)
+* Ceph storage elements should maintain integrity and availability while scaling
+
+### Non-goals
```

Comment:
This section is meant to outline what aren't the goals of the project. It isn't required all of them are security non-goals.
A good example is [Pixie non-goals](https://github.com/cncf/tag-security/blob/main/assessments/projects/pixie/self-assessment.md#non-goals)

Reply:
Expanded here: [ee390db](https://github.com/dhauss/tag-security/commit/ee390db799dfd7245bf65fa995675fe746123939)